### ISSUE TITLE
[Feature request] Show recent news in HTML5 notifications

### ISSUE THREAD

**Issue by C0rn3j** [FreshRSS/FreshRSS#3436]
![image](https://user-images.githubusercontent.com/1641362/107580490-ea2ee200-6bf6-11eb-9db5-efc72896fb79.png)

Currently, the HTML5 notifications are pretty bare with just showing the news count.

I think adding the most recent items (1? 3? Configurable?) would be a great UX improvement.

---
![image](https://user-images.githubusercontent.com/1641362/107581500-6f66c680-6bf8-11eb-9dea-58af17787e6c.png)

How I imagine it could look taking the above feed entry as an example:
``` 
There are 2 new articles to read on FreshRSS:
Firefox releases: Firefox 85.0.2 is now available
... and more!
```

or with changing the current style a bit:
```
Firefox releases: Firefox 85.0.2 is now available
... and 1 more new article to read on FreshRSS!
```

**Comment by Alkarex** [comment]
Could be an interesting option. PR welcome :-)

**Comment by aledeg** [comment]
Might be good to tackle that as well #810

**Comment by C0rn3j** [comment]
How would I go about forcing the notification to appear?
I currently don't know a way which would make it near impossible to make and debug any changes to the logic for me.

**Comment by Alkarex** [comment]
@C0rn3j Open two windows. Mark an article as unread in window 2. In window 1, in the browser console, run `refreshUnreads();`

**Comment by C0rn3j** [comment]
I tried looking into it but I am completely lost, I never worked with any JS project.

From what I gather, `main.js->notifs_html5_show()` needs to be modified to accept an extra variable with the title(s) text.

`main.js->refreshUnreads()` needs to be modified to somehow get the newest unread item(s) and their titles, but this is where I get lost, as that function calls `a=nbUnreadsPerFeed` which _seems_ to be located in `javascriptController.php->nbUnreadsPerFeedAction()` and that seems to currently only get the feeds and if there's any news and not titles/timestamps.

---

Side note, there's an issue when there's only 1 feed to read vs multiple, as '1 new articles' is not grammatically correct, so there should be two versions of the text for singular and plural.

**Comment by Frenzie** [comment]
> Side note, there's an issue when there's only 1 feed to read vs multiple, as '1 new articles' is not grammatically correct, so there should be two versions of the text for singular and plural.

If only it were that easy. ;-) The correct behavior's significantly more complex (albeit not for English).

**Comment by C0rn3j** [comment]
Right, even in my native language (Czech) there'd be three different versions for 1, 2-4 and 5 or more.

Maybe `New articles to read: X` would be grammatically correct in most cases but it looks a bit worse.

**Comment by math-GH** [comment]
Do we know how many characters are allowed to display in the notification message? (Firefox, Chrome etc.?)

**Comment by Frenzie** [comment]
That's more a matter for the OS. Around 50 or 60 for mobile and around 120 for desktop on average I believe. By which I mean you'll see ellipses, I'm not sure how many hundreds or thousands would be the true max.
